### ISSUE TITLE
Error while installing ODISE

### ISSUE THREAD

**Issue by zhizdev** [NVlabs/ODISE#43]
I am running into many errors while installing ODISE. Mainly with compiling Mask2Former. 

Errors include: 
```
 fatal error: 'crypt.h' file not found
```
```
 fatal error: 'cusparse.h' file not found
```

Here is my workaround (or at least attempted). 

Install Mask2former from their repo (https://github.com/facebookresearch/Mask2Former/blob/main/INSTALL.md) -- this is the main issue. However make sure you use python 3.9. 

Once you are able to install Detectron2 and Mask2former. Then you should be set for ODISE. 

I had to append `CUDA_HOME="/usr/local/cuda-11.3" pip install -e .` CUDA_HOME to install detectron2 and mask2former inside my conda environment.

**Comment by zhizdev** [comment]
Also need this to avoid `AttributeError: module 'distutils' has no attribute 'version'`

`pip install setuptools==59.5.0`

**Comment by zhizdev** [comment]
Now running into issues with `ImportError: /home/zhizhuoz/code/mask2former/MultiScaleDeformableAttention.cpython-39-x86_64-linux-gnu.so: undefined symbol: _Z16THPVariable_WrapN2at6TensorE`

**Comment by zhizdev** [comment]
Now running into issues with `ImportError: ~/mask2former/MultiScaleDeformableAttention.cpython-39-x86_64-linux-gnu.so: undefined symbol: _Z16THPVariable_WrapN2at6TensorE`

**Comment by zhizdev** [comment]
nvm closing this 

DO NOT INSTALL Detectron2 and Mask2former from source. Install it with this repo.

**Comment by zhizdev** [comment]
This seems like an important fix `pip install numpy==1.23.1`
